src/internal/connector/sharepoint/data_collections.go

```diff
@@ -71,17 +80,53 @@ func DataCollections(
 				if err != nil {
 					return nil, support.WrapAndAppend(site, err, errs)
 				}
-
-				collections = append(collections, spcs...)
 			}
 
+			collections = append(collections, spcs...)
+
 			foldersComplete <- struct{}{}
 		}
 	}
 
 	return collections, errs
 }
 
+func collectLists(
+	ctx context.Context,
+	serv graph.Servicer,
+	tenantID, siteID string,
+	scope selectors.SharePointScope,
+	updater statusUpdater,
+	ctrlOpts control.Options,
+) ([]data.Collection, error) {
+	logger.Ctx(ctx).With("site", siteID).Debug("Creating SharePoint List Collections")
+
+	if scope.Matches(selectors.SharePointSite, siteID) {
+		spcs := make([]data.Collection, 0)
+
+		dir, err := path.Builder{}.Append(siteID).
+			ToDataLayerSharePointPath(
+				tenantID,
+				siteID,
+				path.ListsCategory,
+				false)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create collection path for site: %s", siteID)
+		}
+
+		collection := NewCollection(dir, serv, updater.UpdateStatus)
```

Comment:
it looks like lists doesn't support delta queries for items, so just doing this and leaving the implementations of functions `State()` and `PreviousPath()` for the collection is fine

Reply:
I stand corrected, the beta API does seem to have a delta endpoint. For now, we can leave this as is and go back and add some extra logic to implement what incrementals will need. There's no reason to squeeze that into this PR especially if we haven't yet started looking at doing incrementals for anything other than Exchange